### PR TITLE
[SDP-663] Publish helm charts for SDP

### PR DESCRIPTION
**Pull request by marwen-abid**
### What

Publish helm charts for SDP. 

* Separate TSS deployments from SDP. 
* Add optional secret creation 
* Remove SDP specific configs

### Why

Publishing helm charts 

### Known limitations

[TODO or N/A]

#### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [ ] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [ ] This PR adds tests for the new functionality or fixes.
* [ ] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [ ] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [ ] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [ ] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [ ] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [ ] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

**Review by marcelosalloum — COMMENTED** [review]
* I've found a few instances of `.Values.service.type`. Should that be moved inside `global`?
* For the dashboard, I've placed its route inside the dashboard hierarchy. Do you think that's a good idea for the rest of the services too? I thought that placing the route inside its own service tag would make more sense:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/802b08f739f212a493d83e5442df48f64a6a2260/helmchart/sdp/values.yaml#L359-L365

**Review by marcelosalloum — COMMENTED** [review]
I also think it'd be better to have the ingress inside its own service, rather than having that shared ingress we have right now (my fault for creating that in the first place). WDYT?

File: helmchart/sdp/templates/NOTES.txt
Comment (marcelosalloum) [inline helmchart/sdp/templates/NOTES.txt:1]: There's a substitution missing for `.Values.service.*` -> `.Values.global.service.*` in line 15:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/b8559ffcf7a89c378a268ea53a8d9dddb96921bb/helmchart/sdp/templates/NOTES.txt#L15

File: helmchart/sdp/values.yaml
Comment (marcelosalloum) [inline helmchart/sdp/values.yaml:1]: I believe `.Values.dashboard.service` could be removed so we rely only on `.Values.global.service`:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/b8559ffcf7a89c378a268ea53a8d9dddb96921bb/helmchart/sdp/values.yaml#L360-L361

**Review by marcelosalloum — APPROVED** [review]
Very cool, thank you so much for working on this!

Feel free to merge it after addressing the comments below. Or let me know if you don't agree with them.

Also, we could update the helm chart version to avoid confusion. The current version is in:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/b8559ffcf7a89c378a268ea53a8d9dddb96921bb/helmchart/sdp/Chart.yaml#L8-L11

File: helmchart/sdp/values.yaml
Comment (marwen-abid) [reply helmchart/sdp/values.yaml:1]: Removed

**Pull request merged**